src/mlir/dialect/VeronaDialect.cc

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "VeronaDialect.h"
+
+#include "VeronaOps.h"
+#include "mlir/IR/DialectImplementation.h"
+#include "mlir/IR/StandardTypes.h"
+
+using namespace mlir;
+using namespace mlir::verona;
+
+//===----------------------------------------------------------------------===//
+// Verona dialect.
+//===----------------------------------------------------------------------===//
+
+VeronaDialect::VeronaDialect(mlir::MLIRContext* context)
+: Dialect(getDialectNamespace(), context)
+{
+  addOperations<
+#define GET_OP_LIST
+#include "dialect/VeronaOps.cpp.inc"
+    >();
+  addTypes<IntegerType>();
+  allowUnknownOperations();
+  allowUnknownTypes();
+}
+
+Type VeronaDialect::parseType(DialectAsmParser& parser) const
+{
+  StringRef keyword;
+  if (parser.parseKeyword(&keyword))
+    return Type();
+
+  if (keyword.startswith("U"))
+  {
+    size_t width = 0;
+    if (keyword.substr(1).getAsInteger(10, width))
+    {
+      parser.emitError(parser.getNameLoc(), "unknown verona type: ") << keyword;
+      return Type();
+    }
+    return IntegerType::get(getContext(), width, false);
+  }
+
+  parser.emitError(parser.getNameLoc(), "unknown verona type: ") << keyword;
+  return Type();
+}
+
+void VeronaDialect::printType(Type type, DialectAsmPrinter& os) const
+{
+  switch (type.getKind())
+  {
+    case VeronaTypes::Integer:
+    {
+      auto iTy = type.cast<IntegerType>();
+      if (iTy.getSign())
+      {
+        os << "S";
+      }
+      else
+      {
+        os << "U";
+      }
+      os << iTy.getWidth();
+      return;
+    }
+    default:
+      llvm_unreachable("unexpected 'verona' type kind");
+  }
+}
+
+//===----------------------------------------------------------------------===//
+// Verona types.
+//===----------------------------------------------------------------------===//
+
+namespace mlir::verona::detail
+{
+  struct IntegerTypeStorage : public ::mlir::TypeStorage
+  {
+    size_t width;
```

Comment:
Sorry, `size_t` is my default (register sized) type, unless I need something else. I'll change to `uint8_t` so that we limit out integer sizes to 2^255, which I think it's good enough.